internal/node/launcher.sh

```diff
@@ -222,13 +220,40 @@ export BAZEL_PATCH_ROOT=$(dirname $PWD)
 # fail before we could assert that we expected that failure.
 readonly EXPECTED_EXIT_CODE="TEMPLATED_expected_exit_code"
 if [ "${EXPECTED_EXIT_CODE}" -eq "0" ]; then
-  # Replace the current process (bash) with a node process.
-  # This means that stdin, stdout, signals, etc will be transparently
-  # handled by the node process.
-  # If we had merely forked a child process here, we'd be responsible
-  # for forwarding those OS interactions.
-  exec "${node}" "${LAUNCHER_NODE_OPTIONS[@]:-}" "${USER_NODE_OPTIONS[@]:-}" "${MAIN}" ${ARGS[@]+"${ARGS[@]}"}
```

Comment:
by not using exec do we still handle "stdin, stdout, signals, etc" still?

Reply:
also if we're ditching `exec` could we compbine this with the one belonw, Line 260

Reply:
Yes. I think combining the two is reasonable. I don't think there is much value in running coverage when expected exit code != 0 but maybe someone will find that useful and it would simplify the code

Reply:
Good point re: the exec. We should look back at why the exec was there (trace back to the commit where it was added and see what it was fixing as I don't recall) and if there is anything special needed to pipe the io streams & signals. The non-exec non-zero exit code path had no special handling but that path is likely not used at all outside of this repo.

Reply:
This looks like where it was introduced https://github.com/bazelbuild/rules_nodejs/commit/c124496339f26e1d4d5bda1fe827220222d9b177

Reply:
Looks like w'ell need to go back to the trap approah to handle SIGTERM, SIGINT, SIGQUIT. will throw that together and see what it looks like with that other refactor in

Reply:
Thanks @Toxicable . Was just looking at that commit and came to the same conclusion.

Reply:
Modified it based off your changes and the ones in that MR, please check it well, my bash skills are not amazing